assessor/lib/state.rb

```diff
@@ -0,0 +1,55 @@
+
+def state
+  # Memoized
+  unless @state
+    # Look for state only in incremental mode. Do not fail if missing,
+    # just fall back to regular mode, starting with empty state.
+    if @incremental && File.exists?(statepath)
+      @state = YAML.load_file(statepath)
+    else
+      @state = {}
+    end
+    # Schema:
+    # - state[<engine>][<version>]['works']	:: boolean
+    # - state[<engine>][<version>]['app']	:: string
+    #
+    # version is chart version  (`version`)
+    # app     is engine version (`appVersion`)
+    @state.default_proc = proc { |h, k| h[k] = Hash.new }
+  end
+  @state
+end
+
+def state_save(engine, enginev, chartv, success)
+  @state[engine] ||= {}
+  @state[engine][chartv] = {
```

Comment:
… as long as you use the accessor here (`state[engine][chartv] = …` without the `@`)

Reply:
Oh. Heh.

Reply:
Fixed with commit dc68b80856848585810d27f16be4ba547b0cd313.